File: 319A.cpp
#include<bits/stdc++.h>
using namespace std;
int main()
{
    string s;
    cin>>s;
    long long a;
    if(s[s.length()-1]=='0')
        a=0;
    else
        a=1;
    reverse(s.begin(), s.end());
    for(int i=1; i<s.length(); i++)
    {
        if(s[i]=='0')
            a = (a*2);
        else
            a = (2*a+(long long)pow(2,2*i));
    }
    cout<<a%1000000007<<endl;
    return 0;
}

File: 319A.py
s = raw_input()
a=0
d = s[::-1]
if d[0]=='0':
    a=0
else:
    a=1
for i in range(1,len(d)):
    if d[i]=='0':
        a*=2
    else:
        a = a*2+pow(2,2*i)
print a%1000000007